ci/release/update-version.sh

```diff
@@ -37,60 +38,83 @@ function sed_runner() {
 
 # Centralized version file update
 echo "${NEXT_FULL_TAG}" > VERSION
+echo "${NEXT_FULL_TAG}" > RAPIDS_VERSION
 
-dependencies='libcuopt libcuopt-cu12 cuopt cuopt-cu12 cuopt-mps-parser'
-for FILE in conda/environments/*.yaml dependencies.yaml; do
-    for dependency in ${dependencies}; do
-        sed_runner "s/\(${dependency}==\)[0-9]\+\.[0-9]\+/\1${PY_NEXT_SHORT_TAG}/" "${FILE}"
-    done
+DEPENDENCIES=(
+  libcuopt
+  libcuopt-cu12
+  cuopt-mps-parser
+  cuopt
+  cuopt-cu12
+  cuopt-server
+  cuopt-server-cu12
+  cuopt-sh-client
+  cudf
+  cudf-cu12
+  cuvs
+  cuvs-cu12
+  librmm
+  librmm-cu12
+  libraft-headers
+  rmm
+  rmm-cu12
+  pylibraft
+  pylibraft-cu12
+  raft-dask
+  raft-dask-cu12
+  rapids-dask-dependency
+)
+
+for DEP in "${DEPENDENCIES[@]}"; do
+  for FILE in dependencies.yaml conda/environments/*.yaml python/*/pyproject.toml; do
+    sed_runner "s/\(${DEP}==\)[0-9]\+\.[0-9]\+/\1${NEXT_SHORT_TAG_PEP440}/" "${FILE}"
+  done
+  for FILE in python/*/pyproject.toml; do
+    sed_runner "s/\(${DEP}==\)[0-9]\+\.[0-9]\+/\1${NEXT_SHORT_TAG_PEP440}/" "${FILE}"
+  done
+  for FILE in docs/cuopt/source/*/quick-start.rst README.md; do
+    sed_runner "s/\(${DEP}==\)[0-9]\+\.[0-9]\+\.\\*/\1${NEXT_SHORT_TAG_PEP440}.\*/g" "${FILE}"
+  done
+  for FILE in docs/cuopt/source/*/quick-start.rst README.md; do
+    sed_runner "s/\(${DEP}=\)[0-9]\+\.[0-9]\+\(\.[0-9]\+\)\?[^ ]*/\1${NEXT_SHORT_TAG}.*/g" "${FILE}"
+  done
+  for FILE in docs/cuopt/source/*/quick-start.rst README.md; do
+    sed_runner "s/\(${DEP}:\)[0-9]\{2\}\.[0-9]\{1,2\}\.[0-9]\+\(-cuda[0-9]\+\.[0-9]\+-\)\(py[0-9]\+\)/\1${DOCKER_TAG}\2\3/g" docs/cuopt/source/cuopt-python/quick-start.rst
+  done
```

Comment:
```suggestion
  for FILE in docs/cuopt/source/*/quick-start.rst README.md; do
    sed_runner "s/\(${DEP}==\)[0-9]\+\.[0-9]\+\.\\*/\1${NEXT_SHORT_TAG_PEP440}.\*/g" "${FILE}"
    sed_runner "s/\(${DEP}=\)[0-9]\+\.[0-9]\+\(\.[0-9]\+\)\?[^ ]*/\1${NEXT_SHORT_TAG}.*/g" "${FILE}"
  done
  sed_runner "s/\(${DEP}:\)[0-9]\{2\}\.[0-9]\{1,2\}\.[0-9]\+\(-cuda[0-9]\+\.[0-9]\+-\)\(py[0-9]\+\)/\1${DOCKER_TAG}\2\3/g" docs/cuopt/source/cuopt-python/quick-start.rst
```

This is currently looping over the same set of files 3 times.

Let's consolidate 2 of those calls so we only loop over the files once, and run that additional line that only applies to  `docs/cuopt/source/cuopt-python/quick-start.rst` outside of the `for FILE` loop, since it only applies to one file.